jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpDateTime.java

```diff
@@ -0,0 +1,229 @@
+import java.util.Objects;
+import java.util.StringTokenizer;
+
+import org.eclipse.jetty.util.Index;
+import org.eclipse.jetty.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HTTP Date/Time parsing and formatting.
+ *
+ * <p>
+ *     Also covers RFC6265 Cookie Date parsing and formatting.
+ * </p>
+ */
+public class HttpDateTime
+{
+    private static final Logger LOG = LoggerFactory.getLogger(HttpDateTime.class);
+
+    private static final Index<Integer> MONTH_CACHE = new Index.Builder<Integer>()
+        .caseSensitive(false)
+        // Note: Calendar.Month fields are zero based.
+        .with("Jan", Calendar.JANUARY + 1)
+        .with("Feb", Calendar.FEBRUARY + 1)
+        .with("Mar", Calendar.MARCH + 1)
+        .with("Apr", Calendar.APRIL + 1)
+        .with("May", Calendar.MAY + 1)
+        .with("Jun", Calendar.JUNE + 1)
+        .with("Jul", Calendar.JULY + 1)
+        .with("Aug", Calendar.AUGUST + 1)
+        .with("Sep", Calendar.SEPTEMBER + 1)
+        .with("Oct", Calendar.OCTOBER + 1)
+        .with("Nov", Calendar.NOVEMBER + 1)
+        .with("Dec", Calendar.DECEMBER + 1)
+        .build();
+
+    private HttpDateTime()
+    {
+    }
+
+    /**
+     * Similar to {@link #parse(String)} but returns unix epoch
+     *
+     * @param datetime the Date/Time to parse.
+     * @return unix epoch in milliseconds, or -1 if unable to parse the input date/time
+     */
+    public static long parseToEpoch(String datetime)
+    {
+        try
+        {
+            Instant instant = parse(datetime);
+            return instant.toEpochMilli();
+        }
+        catch (IllegalArgumentException e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Unable to parse Date/Time: {}", datetime, e);
+            return -1;
+        }
+    }
+
+    /**
+     * <p>Parses a Date/Time value</p>
+     *
+     * <p>Supports the following Date/Time formats found in both
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc9110#name-date-time-formats">RFC 9110 (HTTP Semantics)</a> and
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC 6265 (HTTP State Management Mechanism)</a>
+     * </p>
+     *
+     * <ul>
+     *     <li>{@code Sun, 06 Nov 1994 08:49:37 GMT} - RFC 1123 (preferred)</li>
+     *     <li>{@code Sunday, 06-Nov-94 08:49:37 GMT} - RFC 850 (obsolete)</li>
+     *     <li>{@code Sun Nov  6 08:49:37 1994} - ANSI C's {@code asctime()} format</li>
+     * </ul>
+     *
+     * <p>
+     *  Parsing is done according to the algorithm specified in
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC6265: Section 5.1.1: Date</a>
+     * </p>
+     *
+     * @param datetime a Date/Time string in a supported format
+     * @return an {@link Instant} parsed from the given string
+     * @throws IllegalArgumentException if unable to parse date/time
+     */
+    public static Instant parse(String datetime)
```

Comment:
Can the main method of this class return a ZonedDateTime, then have convenience classes converting that to an epoch or an instance.

Reply:
that doesn't save any code.
in fact it pushes the error handling and epoch handling up into a bunch of other classes that use the parseEpoch right now.

Reply:
Currently you have:
```java
    public static Instant parse(String datetime)
    {
        // ...

        // RFC 6265 - Section 5.1.1 - Step 6
        ZonedDateTime dateTime = ZonedDateTime.of(year,
            month, day, hour, minute, second, 0,
            ZoneId.of("GMT"));

        // RFC 6265 - Section 5.1.1 - Step 7
        return dateTime.toInstant();
    }

    public static long parseToEpoch(String datetime)
    {
        try
        {
            Instant instant = parse(datetime);
            return instant.toEpochMilli();
        }
        catch (IllegalArgumentException e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Unable to parse Date/Time: {}", datetime, e);
            return -1;
        }
    }
```

I'm just saying that it will be more flexible and potentially a tiny bit more efficient to have:

```java
    public static ZonedDateTime parse(String datetime)
    {
        // ...

        // RFC 6265 - Section 5.1.1 - Step 6 & 7
        return ZonedDateTime.of(year,
            month, day, hour, minute, second, 0,
            ZoneId.of("GMT"));
    }

    public static Instance parseToInstant(String datetime)
    {
        // RFC 6265 - Section 5.1.1 - Step 7
        return parse(datetime).toInstant();
    }

    public static long parseToEpoch(String datetime)
    {
        try
        {
            return TimeUnit.SECONDS.toMillis(parse(datetime).toEpochSeconds());
        }
        catch (IllegalArgumentException e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Unable to parse Date/Time: {}", datetime, e);
            return -1;
        }
    }
```

This is a utility class and perhaps somebody wants the ZonedDateTime, so best to make it directly available rather than build one, convert to Instant and then convert back to ZDT with a zone.

Reply:
Got it.  I misunderstood what you were asking for.